File: src/main/python/handler/PersonHandler.py
from Struct import PersonService
from Struct import ttypes
class PersonHandler():
    def __init__(self):
        self.people = {}
    def getPersonByName(self, name):
        if len(self.people) <= 0:
            person = ttypes.Person()
            person.name = "队列为空"
            return person
        if not self.people[name]:
            person = ttypes.Person()
            person.name = "用户["+ name + "]不存在"
            return person
        return self.people[name]

    def addPerson(self, person):
        self.people[person.name] = person

File: src/main/java/com/godme/handler/PersonHandler.java
package com.godme.handler;

import com.godme.thrift.MyException;
import com.godme.thrift.Person;
import com.godme.thrift.PersonService;
import org.apache.thrift.TException;

import java.lang.reflect.Member;
import java.util.HashMap;

public class PersonHandler implements PersonService.Iface {
    HashMap<String, Person> people = new HashMap<>();
    @Override
    public Person getPersonByName(String name) throws MyException {
        if(people.size() <= 0){
            Person person = new Person();
            person.setName("队列为空");
            return person;
        }
        if(!people.containsKey(name)){
            Person person = new Person();
            person.setName("用户"+ name + "不存在");
            return person;
        }
        return people.get(name);
    }

    @Override
    public void addPerson(Person person) throws MyException {
        people.put(person.getName(), person);
    }
}
